src/Microsoft.AspNetCore.OData/Routing/ODataPathExtensions.cs

```diff
@@ -265,6 +265,23 @@ internal static (IEdmProperty, IEdmStructuredType, string) GetPropertyAndStructu
                 return (null, structuredType, name);
             }
 
+            if (segment is OperationImportSegment operationImportSegment)
+            {
+                IEdmOperationImport operationImport = operationImportSegment.OperationImports.First();
+                IEdmTypeReference edmType = operationImport.Operation.ReturnType;
+                if (edmType == null)
+                {
+                    return (null, null, operationImport.Name);
+                }
+
+                if (edmType.IsCollection())
+                {
+                    edmType = edmType.AsCollection().ElementType();
+                }
```

Comment:
Can this be simplified using `AsElementType()`?

```suggestion
                edmType = edmType.Definition.AsElementType();
```

Sincde `AsCollection()` also checks whether this `edmType` is a collection, this could also avoid redundant checks.